raiden/tests/unit/transfer/test_source_routing.py

```diff
@@ -79,6 +82,30 @@ def test_metadata_hashing():
     assert one_hash != inverted_route_hash, "route metadata with inverted routes still match"
 
 
+def test_route_metadata_displayname_validation():
+    properties = factories.RouteMetadataProperties()
+    one_metadata = factories.create(properties)
+    route = one_metadata.route
+    signer = factories.make_signer()
+    server_name = urlparse("https://ownserver.com").netloc
+    user_id = f"@{to_normalized_address(signer.address)}:{server_name}"
+    displayname = encode_hex(signer.sign(user_id.encode()))
+
+    bad_metadata = RouteMetadata(
+        route=route,
+        address_metadata={signer.address: {"user_id": user_id, "displayname": "fake"}},
+    )
+    good_metadata = RouteMetadata(
+        route=route,
+        address_metadata={signer.address: {"user_id": user_id, "displayname": displayname}},
+    )
+
+    assert bad_metadata.address_metadata == {}
+    assert good_metadata.address_metadata == {
+        signer.address: {"user_id": user_id, "displayname": displayname}
+    }
+
+
```

Comment:
Maybe a test where a false verifaication of displayname would remove the metadata would be nice to be covered here

Reply:
That's what lines 94 and 103 are about. Or did you have something different in mind?

Reply:
my bad. I dont know what I saw there before...